eng/common/scripts/Package-Properties.ps1

```diff
@@ -79,6 +82,61 @@ class PackageProps
         $this.Initialize($name, $version, $directoryPath, $serviceDirectory)
         $this.Group = $group
     }
+
+    hidden [object]GetValueSafely($hashtable, [string[]]$keys) {
+        $current = $hashtable
+        foreach ($key in $keys) {
+            if ($current.ContainsKey($key) -or $current[$key]) {
+                $current = $current[$key]
+            }
+            else {
+                return $null
+            }
+        }
+
+        return $current
+    }
+
+    hidden [HashTable]ParseYmlForArtifact([string]$ymlPath) {
+        if (Test-Path -Path $ymlPath) {
+            try {
+                $content = Get-Content -Raw -Path $ymlPath | CompatibleConvertFrom-Yaml
+                if ($content) {
+                    $artifacts = $this.GetValueSafely($content, @("extends", "parameters", "Artifacts"))
+
+                    $artifactForCurrentPackage = $artifacts | Where-Object { $_["name"] -eq $this.ArtifactName -or $_["name"] -eq $this.Name }
+
+                    if ($artifactForCurrentPackage) {
+                        return [HashTable]$artifactForCurrentPackage
+                    }
+                }
+            }
+            catch {
+              Write-Host "Exception while parsing yml file $($ymlPath): $_"
+            }
+        }
+
+        return $null
+    }
+
+    [void]InitializeCIArtifacts(){
```

Comment:
Before someone says

> you forgot to add `hidden` to this one too Scott

This is because `azure-sdk-for-js` sets `artifactName` _after_ they create the package prop class instance for the first time, so their `Get-PackagePropertiesFromRepo` needs to _call_ `InitializeCIArtifact` to populate the ArtifactDetails field.